File: backend-spring-boot/src/main/java/com/org/backend/models/Position.java
package com.org.backend.models;

import java.io.Serializable;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@Data
@Entity
@Table(name = "position")
public class Position implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.UUID)
	private String id;
	@Column(nullable = false, unique = true, length = 30)
	private String position;
	@Column(nullable = true)
	private Double bonus;
	@JsonIgnore
	@ToString.Exclude
	@EqualsAndHashCode.Exclude
	@OneToMany(mappedBy = "position")
	private List<Employee> employees;
}

File: frontend-react/src/models/Position.ts
export default class Position {
  private id: string;
  private position: string;
  private bonus: number;

  constructor(id: string, position: string, bonus: number) {
    this.id = id;
    this.position = position;
    this.bonus = bonus;
  }

  getPosition() {
    return {
      id: this.id,
      position: this.position,
      bonus: this.bonus,
    };
  }
}
